### PR TITLE
Cache Typst packages

### PR DESCRIPTION
**Pull request by yusancky**

### CODE REVIEW

**Comment by yusancky** [comment]
@jcbhmr I have no experience with Vite and suspect that the error might be related to my usage of it. Could you please offer some assistance?

**Review by jcbhmr — COMMENTED** [review]
I think you need to use `glob.hashFiles()` across any files like `package-lock.json` or in this case `**/*.typ` to get a unique cache id. Then save to _that_ instead of `packages-id`? Similar to https://github.com/actions/setup-node ? does that make sense?

as for the "vite doesn't work" you're absolutely 100% right. 😆 it workd great up until it didn't. https://bun.sh seems to work in this instance so i guess you can use that instead

i made a bunch of changes in #17 that i think you should check out and consider merging into this #16 branch 🤷

**Comment by yusancky** [comment]
> I think you need to use `glob.hashFiles()` across any files like `package-lock.json` or in this case `**/*.typ` to get a unique cache id. Then save to _that_ instead of `packages-id`? Similar to https://github.com/actions/setup-node ? does that make sense?

I had considered this option. But since Typst doesn't currently generate package lists or support installing packages in a list, my idea at the time was for the user to leave a `requirements.typ`, contains the `#import` for the packages needed to use, then downloads packages and saves them to the cache.

I was thinking about whether there were other solutions, but given that some of the packages had dependencies on other packages, etc., I asked @Enter-tainer for advice privately. He suggested I just cache the paths directly. To prevent caching too much from long term use, I implemented the option, and it resulted in the current setup.

I am open to your suggestions for improvement. I wonder if you have any new insights on this matter, and if there might be any potential new bottlenecks that could arise as a result? I would love to hear your thoughts.

File: src/main.ts [hunk 1]
```diff
@@ -71,3 +71,24 @@ if (!found) {
...
 core.info(`✅ Typst v${version} installed!`);
+
+let packagesId = core.getInput("packages-id");
+if (!packagesId || packagesId === "-1") {
+  core.info(`No packages should be installed.`);
+} else {
+  const cachePath = {
+    darwin: "~/Library/Caches",
+    linux: "$XDG_CACHE_HOME",
+    win32: "~/Library/Caches",
```
Comment (Enter-tainer) [inline src/main.ts:82]: typo?

File: src/post.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+let packagesId = core.getInput("packages-id");
+
+if (!packagesId || packagesId === "-1") {
+  core.info("No packages should be cached.");
+} else {
+  const cachePath = {
+    darwin: "~/Library/Caches",
+    linux: "$XDG_CACHE_HOME",
+    win32: "~/Library/Caches",
```
Comment (yusancky) [inline src/post.ts:13]:
```suggestion
    win32: "%LOCALAPPDATA%",
```

File: src/main.ts (hunk 1 above)
Comment (yusancky) [inline src/main.ts:82]:
```suggestion
    win32: "%LOCALAPPDATA%",
```

File: README.md [hunk 3]
```diff
@@ -12,7 +12,7 @@
 </table>
 
 📝 Installs [Typst] for GitHub Actions \
-⚡ Caches installation files
+⚡ Caches installation files and packages
```
Comment (yusancky) [inline README.md:15]:
```suggestion
⚡ Caches Typst installation in the tool cache \
📦 Caches [packages](https://github.com/typst/packages) as dependencies
```

**Comment by yusancky** [comment]
Test: https://github.com/typst-community/setup-typst/actions/runs/7766346043

CC: @jcbhmr @Enter-tainer

**Comment by yusancky** [comment]
CC: @jcbhmr

**Comment by yusancky** [comment]
I don't know why sometimes it cannot stop normally, but sometimes it can.

CC: @jcbhmr

**Comment by yusancky** [comment]
@jcbhmr I think this can be merged into the main branch, what's your opinion?

**Pull request merged**